getMac changed to getUserName that uses sharedPreferences

teamagam1 opened the pull request:
getMac changed to getUserName that uses sharedPreferences

teamagam1 committed:
getMac changed to getUserName that uses sharedPreferences

review-ninja commented:
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/52" target="_blank"><img src="https://app.review.ninja/52010471/pull/52/badge" alt="ReviewNinja"/>

BarAgam commented on app/src/main/java/com/teamagam/gimelgimel/app/utils/NetworkUtil.java:
```diff
@@ -96,7 +98,8 @@ public static InetAddress getIp() {
-        return sMacAddress;
+    public static String getUserName(Context context) {
+        PreferenceUtil prefs = ((GGApplication) context.getApplicationContext()).getPrefs();
+        return prefs.getString(R.string.user_name_text_key,sMacAddress);
```
It seems that a user name is always available since the user name preference has a default value (strings.pref_default_display_name).
If that's the situation, all the mac-address code isn't necessary, and can be safely deleted. !resolve

teamagam1 replied:
!resolved

BarAgam commented on app/src/main/java/com/teamagam/gimelgimel/app/utils/NetworkUtil.java:
```diff
@@ -96,7 +98,8 @@ public static InetAddress getIp() {
 
-    public static String getMac() {
-        return sMacAddress;
+    public static String getUserName(Context context) {
```
See the comment below before reading...

This method is currently at NetowrkUtils, which makes no sense.
If all of the MAC logic is removed as I suggested at the comment below, the util class can be renamed. 
Otherwise, It should be moved to more suiting place !resolve

teamagam1 replied:
!resolved

BarAgam commented on app/src/main/java/com/teamagam/gimelgimel/app/network/services/GGMessagingUtils.java:
```diff
@@ -1,5 +1,6 @@
+import android.content.Context;
 import android.util.Log;
 
 import com.teamagam.gimelgimel.app.model.ViewsModels.Message;
```
I;m having a hard time deciding myself, so I'm sharing my thoughts:

This class seems like it should be refactored, which will effect other parts of the codebase.

The problem right now is that all of these utility classes share the same structure, and it can be resolved by refactoring it to a GGMessageSender class (or whatever name you see fit) that will expose all the needed functionality (different type of messages sending).

It wont be a huge change, but I think it would clean up the code a bit, and stop this (soon to be) monstrous class evolutions

teamagam1 replied:
private member
register to update of user name 
constructor with context

teamagam1 replied:
!fix

teamagam1 replied:
!fixed

teamagam1 committed:
getuserName moved from NetworlUtil to PreferneceUtil.

rem NetworkUtil (no need for the MAC address anymore)

BarAgam commented on app/src/main/java/com/teamagam/gimelgimel/app/utils/PreferenceUtil.java:
```diff
@@ -23,6 +27,17 @@ public PreferenceUtil(Resources res, SharedPreferences preferences) {
     }
 
+    /*
+     * some static util functions
```
Code pollution

BarAgam replied:
Either write a meaningful comment (a comment that explains something the code fails to pass) or don't write one at all !fix

teamagam1 replied:
!fixed

BarAgam commented on app/src/main/java/com/teamagam/gimelgimel/app/utils/PreferenceUtil.java:
```diff
@@ -23,6 +27,17 @@ public PreferenceUtil(Resources res, SharedPreferences preferences) {
+    }
+    /*
+
+     */
```
empty comment !resolve

teamagam1 replied:
!resolved

gRaytan committed:
GGMessaingUtils changed name to GGMessageSender and is not static anymore.

BarAgam commented on app/src/main/java/com/teamagam/gimelgimel/app/control/receivers/NewLocationBroadcastReceiver.java:
```diff
@@ -23,7 +23,8 @@ public void onReceive(Context context, Intent intent) {
                     LocationFetcher.KEY_NEW_LOCATION_SAMPLE);
             Log.v(LOG_TAG, "New Location: " + loc.toString());
-            GGMessagingUtils.sendUserLocationMessageAsync(loc);
+            GGMessageSender messageSender = new GGMessageSender(context);
```
Make GGMessageSender a dependency for this class.
Meaning, it gets an instance at the constructor.
Use can use a static factory method to instantiate one from context. !fix

teamagam1 replied:
!fixed

BarAgam commented on app/src/main/java/com/teamagam/gimelgimel/app/network/services/GGMessageSender.java:
```diff
@@ -20,20 +24,31 @@
+    private final PreferenceUtil mPrefs;
+    private String mSenderId;
+
+    public GGMessageSender(Context context) {
```
Can be made static with context argument.
This way, getUsername will be usable without constructing GGMessageSender. !fix

BarAgam replied:
!fixed
Ignore... comment misplaced. meant it to be placed at getUserName

BarAgam commented on app/src/main/java/com/teamagam/gimelgimel/app/network/services/GGMessageSender.java:
```diff
@@ -20,20 +24,31 @@
+        mPrefs.registerOnSharedPreferenceChangeListener(this);
+    }
+
+    public String getUserName() {
```
Can be made static with context argument.
This way, getUsername will be usable without constructing GGMessageSender. !fix

teamagam1 replied:
!fixed

teamagam1 committed:
Merge branch 'feature/map_related_bugs' into task/map_related_bugs/cesium_load_error

teamagam1 committed:
CR fixes

BarAgam commented on app/src/main/java/com/teamagam/gimelgimel/app/network/services/GGMessageSender.java:
```diff
@@ -20,20 +24,28 @@
+        mPrefs.registerOnSharedPreferenceChangeListener(this);
+    }
+
+    public static String getUserName(Context context) {
```
Reorganize static method position !fix

BarAgam replied:
!fixed

barmako committed:
Refactored GGMessageSender:
 1. Encapsulated PreferenceChangedListener.
 2. Reorganized static public method to meet conventions
 3 Converted a field to local variable
